app/src/main/java/com/infomaniak/swisstransfer/ui/screen/onboarding/OnboardingScreen.kt

```diff
@@ -0,0 +1,195 @@
+        }
+    }
+
+    OnboardingScaffold(
+        pagerState = pagerState,
+        onboardingPages = onboardingPages,
+        bottomContent = {
+            BottomContent(
+                modifier = Modifier
+                    .padding(it)
+                    .consumeWindowInsets(it),
+                isLastPage = { isLastPage },
+                startMainActivity = goToMainActivity,
+                goToNextPage = { coroutineScope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) } },
+            )
+        },
+    )
+}
+
+@Composable
+private fun Page.toOnboardingPage(isHighlighted: Map<Page, MutableState<Boolean>>) = OnboardingPage(
+    background = background.image(),
+    illustration = { Illustration(illustration) },
+    text = {
+        TitleAndDescription(
+            page = this,
+            isHighlighted = { isHighlighted[this]?.value ?: false }
+        )
+    }
+)
+
+@Composable
+private fun Illustration(illustration: ThemedImage) {
+    Image(illustration.image(), contentDescription = null)
+}
+
+@Composable
+private fun TitleAndDescription(page: Page, isHighlighted: () -> Boolean) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+        Text(
+            textAlign = TextAlign.Center,
+            text = stringResource(page.titleRes),
+            style = SwissTransferTheme.typography.specificLight22,
+        )
+
+        Spacer(modifier = Modifier.height(Margin.Mini))
+
+        HighlightedText(
+            modifier = Modifier.widthIn(max = Dimens.DescriptionWidth),
+            templateRes = page.subtitleTemplateRes,
+            argumentRes = page.subtitleArgumentRes,
+            style = SwissTransferTheme.typography.h1,
+            angleDegrees = page.highlightAngleDegree,
+            isHighlighted = isHighlighted,
+        )
+    }
+}
+
+@Composable
+private fun BottomContent(
+    modifier: Modifier = Modifier,
+    isLastPage: () -> Boolean,
+    startMainActivity: () -> Unit,
+    goToNextPage: () -> Unit
+) {
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = modifier
+            .fillMaxWidth()
+            .height(140.dp),
+    ) {
+        AnimatedOnboardingButton(
+            isExpanded = { isLastPage() },
+            onClick = { if (isLastPage()) startMainActivity() else goToNextPage() },
+        )
+    }
+}
+
+private enum class Page(
+    val background: ThemedImage,
+    val illustration: ThemedImage,
+    @StringRes val titleRes: Int,
+    @StringRes val subtitleTemplateRes: Int,
+    @StringRes val subtitleArgumentRes: Int,
+    val highlightAngleDegree: Double,
+) {
+    STORAGE(
+        background = AppIllus.RadialGradientCornerTopRight,
+        illustration = AppIllus.StorageBoxPile,
+        titleRes = R.string.onboardingStorageTitle,
+        subtitleTemplateRes = R.string.onboardingStorageSubtitleTemplate,
+        subtitleArgumentRes = R.string.onboardingStorageSubtitleArgument,
+        highlightAngleDegree = -3.0,
```

Comment:
We could extract this to a private const too, especially as all the angle of these highlighted texts should probably always stay the same for this case